docs/_releases/v2.0.0-pre.4/matchers.md

```diff
@@ -112,6 +112,26 @@ Requires the value to be a `Function`.
 Requires the value to be an `Array`.
 
 
+#### `sinon.match.array.deepEquals(iterable)`
+
+Requires an iterable value to be deep equal another one.
```

Comment:
This is a tricky question, but according to MDN this is the definition of an `iterable`:

>In order to be iterable, an object must implement the `@@iterator` method, meaning that the object (or one of the objects up its prototype chain) must have a property with a `@@iterator` key which is available via constant `Symbol.iterator`

Also there are lots of objects other than arrays that can be considered iterables, such as Strings, TypedArrays, Maps and Sets. The `arguments` object is also considered an `iterable`.

I think that maintaining these matchers under the `array` property will cause confusion and will not leave it explicit that it can be used with other objects (Maps, Sets, etc...).

I'd like to change this as suggested by @fearphage.

Please let me know what you think about this subject. Also, if we agree on changing this, would you like me to address that in a separate PR and them amend a commit for fixing this one or would you like me to rename this PR and work on this branch?